Add --enable-strmax configuration option support

psmitsu opened the pull request:
Specify library name and version:  **libsafec/3.6.0**

Hello! This PR enables setting "--enable-strmax" configure flag through Conan package option, as described in #14611 

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/conan-io/conan-center-index?pullRequest=14612) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/conan-io/conan-center-index?pullRequest=14612) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/conan-io/conan-center-index?pullRequest=14612) it.</sub>

jwillikers commented on recipes/libsafec/all/conanfile.py:
```diff
@@ -84,6 +86,7 @@ def _configure_autotools(self):
             "--enable-debug={}".format(yes_no(self.settings.build_type == "Debug")),
             "--disable-doc",
             "--disable-Werror",
+            "--enable-strmax={}".format(self.options.strmax),
```
What happens when `self.options.strmax` is `None`?

jwillikers reviewed (dismissed):
LGTM

uilianries commented:
@psmitsu Thank you for your contribution, please, consider https://github.com/psmitsu/conan-center-index/pull/1

psmitsu commented:
@jwillikers @uilianries thank you for your feedback! I've merged that PR: it had "del rm_safe()" typos which I've fixed. I've also added a commit which allows to set "--enable-memmax" option similarly to "--enable-strmax". Is it okay to do that in the same PR? I thought that it's better to pack changes in one PR, based on @uilianries PR.

uilianries commented:
@psmitsu It's. BTW have you asked for access on the issue #4 already? It's mandatory to the CI run your PR.

prince-chrismc reviewed (approved):
There a few things for the 2.0 migration which would be a great follow up PR :)

but this looks really good start!

prince-chrismc commented:
> Is it okay to do that in the same PR? 

The less PRs the better for the most part, any update to the same recipe can be in the same PR :)
